deploy/eck-stack/examples/fleet-agents.yaml

```diff
@@ -0,0 +1,208 @@
+    #
+    config:
+      node.store.allow_mmap: false
+
+eck-kibana:
+  enabled: true
+
+  # Name of the Kibana instance.
+  #
+  fullnameOverride: kibana
+  
+  # Version of Kibana.
+  #
+  version: 8.2.3
+  
+  spec:
+    # Count of Kibana instances to create.
+    #
+    count: 1
+  
+    # Reference to ECK-managed Elasticsearch instance, ideally from {{ "elasticsearch.fullname" }}
+    #
+    elasticsearchRef:
+      name: elasticsearch
+
+    config:
+      # Note that these are specific to the namespace into which this example is installed, and are
+      # using `default` as configured here.  If installed outside of the `default` namespace,
+      # these 2 lines need modification.
+      xpack.fleet.agents.elasticsearch.hosts: ["https://elasticsearch-es-http.default.svc:9200"]
+      xpack.fleet.agents.fleet_server.hosts: ["https://fleet-server-agent-http.default.svc:8220"]
+      xpack.fleet.packages:
+      - name: system
+        version: latest
+      - name: elastic_agent
+        version: latest
+      - name: fleet_server
+        version: latest
+      - name: kubernetes
+        version: latest
+      xpack.fleet.agentPolicies:
+      - name: Fleet Server on ECK policy
+        id: eck-fleet-server
+        namespace: default
+        monitoring_enabled:
+        - logs
+        - metrics
+        is_default_fleet_server: true
+        package_policies:
+        - name: fleet_server-1
+          id: fleet_server-1
+          package:
+            name: fleet_server
+      - name: Elastic Agent on ECK policy
+        id: eck-agent
+        namespace: default
+        monitoring_enabled:
+        - logs
+        - metrics
+        unenroll_timeout: 900
+        is_default: true
+        package_policies:
+        - package:
+            name: system
+          name: system-1
+        - package:
+            name: kubernetes
+          name: kubernetes-1
+
+eck-agent:
+  enabled: true
+
+  # Version of Elastic Agent.
+  #
+  version: 8.2.3
+  
+  spec:
+    # This is required to remove the default configuration for fleet agent in the default values.yaml
+    # Also note that there is a bug preventing the nullification of child chart's maps, which causes
+    # issues when templating this sub-chart.
+    # PR to resolve here: https://github.com/helm/helm/pull/9138
+    config: null
+
+    # Reference to ECK-managed Kibana instance.
+    #
+    kibanaRef:
+      name: kibana
+
+    elasticsearchRefs: []
+
+    # Reference to ECK-managed Fleet instance.
+    #
+    fleetServerRef:
+      name: fleet-server
+    
+    mode: fleet
+
+    daemonSet:
+      podTemplate:
+        spec:
+          serviceAccountName: elastic-agent
+          hostNetwork: true
+          dnsPolicy: ClusterFirstWithHostNet
+          automountServiceAccountToken: true
+          securityContext:
+            runAsUser: 0
+  
+  serviceAccount:
+    name: elastic-agent
+
+  clusterRole:
```

Comment:
Same as the example in the Agent chart: the `clusterRole` and `clusterRoleBinding` is not necessary here because we have a sensible default for it.